### DISCUSSION TITLE
n8n env file help

### DISCUSSION THREAD

**Discussion by alfiro** [tteck/Proxmox#1801] [Help Needed]
Hi all,
I just install with the script the n8n LXC.
I have error running the telegram trigger and i discovered that it is because is taking localhost as ip address. i think that can be changed in the env file but I don't know how to access that file.
Any ideas?
If I connect with ssh to my LXC it ask user and password and don't know both. Which are the ssh credentials?
Thank you in advance for your help.

**Comment by rafaeljuniorvip** [comment]
Você pode editar o arquivo e definir o caminho .env 
por padrão, a instalação não utiliza o .env

altere no seguinte:
nano /etc/systemd/system/n8n.service

EnvironmentFile=/root/.n8n/.env

E então edite esse arquivo. A configuração do .env pode ser encontrada em outros sites, segue o padrão já utilizado pelo n8n.

Se você precisar de um arquivo .env para configurar o n8n, a primeira etapa seria decidir quais configurações são necessárias para a sua instalação. Depois, crie o arquivo .env no local apropriado (por exemplo, no diretório de trabalho do n8n que você deseja usar ou em um local que você especificar via EnvironmentFile no serviço systemd) e insira as configurações lá.

As variáveis de ambiente comuns que você pode querer configurar incluem:

N8N_BASIC_AUTH_ACTIVE: para ativar a autenticação básica.
N8N_BASIC_AUTH_USER: o nome de usuário para autenticação básica.
N8N_BASIC_AUTH_PASSWORD: a senha para autenticação básica.
DB_TYPE: o tipo de banco de dados (por exemplo, postgresdb, mysqldb).
DB_POSTGRESDB_HOST: o host do banco de dados PostgreSQL.
DB_POSTGRESDB_DATABASE: o nome do banco de dados PostgreSQL.
DB_POSTGRESDB_USER: o usuário do banco de dados PostgreSQL.
DB_POSTGRESDB_PASSWORD: a senha do banco de dados PostgreSQL.
Lembre-se de adaptar o caminho e as variáveis conforme necessário para o seu ambiente específico.

**Comment by erreib** [comment]
I am having a similar problem. I need to set the webhook URL so I tried both the documentation on https://docs.n8n.io/hosting/configuration/configuration-examples/webhook-url/ and the above instructions on changing configuring an .env file, but nothing seems to let me change actual value. Does anyone have any ideas? It's not clear to me whether or not n8n needs to be rebuilt after or how.

**Reply by pythongod** [reply]
See here https://github.com/tteck/Proxmox/issues/1600#issuecomment-1647006540

**Comment by pythongod** [comment]
https://github.com/tteck/Proxmox/issues/1600#issuecomment-1647006540 this is the solution
